app/controllers/main_controller.rb

```diff
@@ -182,4 +182,13 @@ def get_blank_message(login, password)
   def implicit_authorization_target
     OpenStruct.new policy_class: MainPolicy
   end
+
+  def resolve_layout
```

Comment:
Adding this method shouldn't be necessary. You can specify the layout explicitly in a `render` call by adding `layout: ...` as an argument.